packages/api-client/src/components/ApiClient/Request/RequestAuthentication/SecurityScheme.vue

```diff
@@ -190,6 +199,74 @@ function authorizeWithPassword(
     })
 }
 
+function authorizeWithClientCredentials(
+  tokenUrl?: string,
+  options?: {
+    baseUrl?: string
+    proxy?: string
+  },
+) {
+  // Require tokenUrl
+  if (typeof tokenUrl !== 'string') {
+    console.log('tokenUrl is not a string')
+    return
+  }
+
+  // Append baseUrl to tokenUrl if it’s a relative path
+  const urlAndPath =
+    options?.baseUrl && !tokenUrl.startsWith('http')
+      ? concatenateUrlAndPath(options?.baseUrl, tokenUrl)
+      : tokenUrl
+
+  // POST https://api.authorization-server.com/token
+  const url = new URL(urlAndPath)
+
+  // grant_type=client_credentials&
+  // client_id=CLIENT_ID&
+  // client_secret=CLIENT_SECRET
+  const formData = new URLSearchParams()
+  formData.set('grant_type', 'client_credentials')
+  formData.set('client_id', authentication.oAuth2.clientId)
+  formData.set('client_secret', authentication.oAuth2.clientSecret)
+  formData.set('scope', authentication.oAuth2.scopes.join(' '))
+
+  // TODO: Proxy this request
+  fetch(
+    options?.proxy
+      ? redirectToProxy(options?.proxy, url.toString())
+      : url.toString(),
+    {
+      method: 'POST',
+      headers: {
+        'Authorization': `Basic ${btoa(`${authentication.oAuth2.clientId}:${authentication.oAuth2.clientSecret}`)}`,
```

Comment:
Currently have this as both basic auth and in the post form body as both are accepted for client credentials. Not sure if having both will cause problems for some implementations of the flow, so could potentially have some form of switch to pick between the two.